### PR TITLE
Issue #374: Fix SingleBreakOrContinue violations

### PR DESCRIPTION
**Pull request by yaziza**

### CODE REVIEW

**Comment by yaziza** [comment]
Update: 6/7 violations resolved.

Im still working on the last violation. After fixing i will extend the unit tests to meet the requirements.

File: sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/CustomDeclarationOrderCheck.java [hunk 1]
```diff
@@ -596,37 +596,27 @@ private int getPositionInOrderDeclaration(final DetailAST ast)
...
-                }
-                else if (currentRule.hasRule(MAIN_METHOD_MACRO)) {
-                    if (isMainMethod(ast)) {
+
+                    if (isAnonymousClassField(ast)
+                            || classDetail.containsGetter(methodName)
+                            || classDetail.containsSetter(methodName)
+                            || isMainMethod(ast)) {
                         result = index;
                         break;
```
Comment (yaziza) [inline sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/CustomDeclarationOrderCheck.java]: After reducing the number of breaks to 2, not sure how to eleminate the last one ...
Maybe completly refactoring this method ...

**Comment by romani** [comment]
please remove from you commit all unrelated changes(that are just formatting), or move them in separate commit of this PR. This will ease diff verification.

File: sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/CustomDeclarationOrderCheck.java (hunk 1 above)
Comment (romani) [inline sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/CustomDeclarationOrderCheck.java]: code is not simple :), this is just good example that need to follow new Checks ourself.

solution is to put `result != null` condition to FOR, and remove `break` completely.

**Comment by romani** [comment]
Travis is not happy with reduced coverage
> [ERROR] Jan 11, 2016 7:59:02 PM net.sourceforge.cobertura.coveragedata.CoverageDataFileHandler loadCoverageData
INFO: Cobertura: Loaded information on 63 classes.
com.github.sevntu.checkstyle.checks.coding.DiamondOperatorForVariableDefinitionCheck failed check. Branch coverage rate of 85.0% is below 90.0%

It might be resulted by removing the code lines. Please generate coverage report and share pictures of coverage before and after for this method. If it is the same it is ok to change configuration(see cobertura section `pom.xml`) there limit is specified.

File: sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/CustomDeclarationOrderCheck.java (hunk 1 above)
Comment (yaziza) [inline sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/CustomDeclarationOrderCheck.java]: Sorry, i didn't get it. 

What about setting: index = Integer.MAX_VALUE - 1; instead f break ?

Comment (romani) [inline sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/CustomDeclarationOrderCheck.java]: make loop like `for (int index = 0; index < customOrderDeclaration.size() && result != 1 ; index++) `
no breaks are required.

**Comment by romani** [comment]
@yaziza , 1.18 release will be this Sunday, please find time to finish this PR.

**Comment by yaziza** [comment]
Fixed, i have added a new case in the test input file and the branch cover is now 94%
Also i have removed the unnesseray != null check, which will be checked anyway in equals.

Thank you for your support !

Do you have any suggestions on issues i can tackle next ?

**Comment by romani** [comment]
merged as FF. Thanks a lot for your time and contributions.

> Also i have removed the unnesseray != null check

One of the banefit of 100%  coverage is to remove dead/extra code. 

> Do you have any suggestions on issues i can tackle next ?

Can you help us with #410 ? Exceptions during execution usually force user to deactivate the Check at all. But you can choose any unassigned issue.

**Pull request closed without merging**